src/crypto/CHIPCryptoPAL.h

```diff
@@ -70,8 +71,10 @@ constexpr size_t kMAX_Point_Length           = kP256_Point_Length;
 constexpr size_t kMAX_Hash_Length            = kSHA256_Hash_Length;
 
 // Max CSR length should be relatively small since it's a single P256 key and
-// no metadata is expected to be honored by the CA.
-constexpr size_t kMAX_CSR_Length = 255;
+// no metadata is expected to be honored by the CA. This size is an implementation
+// compromise and is not spec-mandated. It is sufficiently large for a PKCS#10
+// CSR that contains approximately 150 bytes of ASN.1 extensions.
```

Comment:
This seems like it would lead to interop problems if some commissioners allow extensions and others do not... so some nodes can only be commissioned by some commissioners.

Yes, if all the commissioners are using the SDK that's great, but they could be using old SDK versions too....

This seems like it needs a spec issue to clearly define the expectations here.  But if the expectation is that any NOCSRElements the spec allows must be allowed, then the spec allows that to be up to 900 bytes.  That means the csr itself can be up to `900 - 1 - 2 - 2 - 32 - 1` bytes (if there are no vendor_reserved bits, 1 byte for struct start, 2 bytes control + tag, then the csr, then 2 bytes control+tag, 32 bytes nonce, one byte struct end).

Reply:
Technically neither GenerateCertificateSigningRequest() nor VerifyCertificateSigningRequestFormat() should be limited to kMAX_CSR_Length whatever that value is decided to be. Both routines are getting a buffer limited by the caller, so if the caller passes in a 900 - 1 - 2 - 2 - 32 - 1 byte buffer then that is the limit. The constant kMAX_CSR_Length should not be used in CHIPCryptoPAL.cpp at all. Regarding interop, extended csr would of course only be used if controller and device acknowledged support for it by any means.

Reply:
Will see if there is a way to remove this. While it's true the caller provides the buffer, there is, everywhere it's used, a call sequence. That call sequence today in the SDK allocates stack buffers. I can probably move the CSR down to size where it's created.

Reply:
So what I did in https://github.com/project-chip/connectedhomeip/pull/28899/commits/f835837bafcca8b2631455708adcd19c5f53c7b7 is to remove the offending checks for max sizes. Now we have a min size required to fit, and since the caller passes the buffer, as @bluebin14 indicated, this should work everywhere.